### PR TITLE
feat: add max beds and house client id

### PR DESCRIPTION
**Pull request by dex-89**
# What's added:
- [x] Convert `server_id` to `client_id` (supported canary map)

**New options in house's window :**
- [x] `Client ID`
- [x] `Max beds`

image

### CODE REVIEW

**Comment by Aerwix** [comment]
When it's going to happen this merge? :P

File: source/iomap_otbm.cpp [hunk 1]
```diff
@@ -1179,6 +1179,14 @@ bool IOMapOTBM::loadHouses(Map& map, pugi::xml_document& doc)
 			warning("House %d has no town! House was removed.", house->id);
 			map.houses.removeHouse(house);
 		}
+
+		if((attribute = houseNode.attribute("clientid"))) {
+			house->clientid = pugi::cast<int32_t>(attribute.value());
+		}
+
+		if((attribute = houseNode.attribute("beds"))) {
+			house->beds = pugi::cast<int32_t>(attribute.value());
```
Comment (luanluciano93) [inline source/iomap_otbm.cpp:1188]:
```suggestion
			house->beds = attribute.as_uint();
```

File: source/iomap_otbm.cpp [hunk 2]
```diff
@@ -1179,6 +1179,14 @@ bool IOMapOTBM::loadHouses(Map& map, pugi::xml_document& doc)
 			warning("House %d has no town! House was removed.", house->id);
 			map.houses.removeHouse(house);
 		}
+
+		if((attribute = houseNode.attribute("clientid"))) {
+			house->clientid = pugi::cast<int32_t>(attribute.value());
```
Comment (luanluciano93) [inline source/iomap_otbm.cpp:1184]:
```suggestion
			house->clientid = attribute.as_uint();
```

**Comment by luanluciano93** [comment]
Com as alterações que eu indiquei, funcionou ... testei mudando o valor total das camas e salvando o mapa ...
image

Posterior abrindo novamente e o valor estava correto.

image

**Comment by luanluciano93** [comment]
Fiz um teste no jogo, não sei se tem alguma relação .. mas define o máximo de camas 2.

image

**Comment by luanluciano93** [comment]
@dex-89, excuse my ignorance, but what is this "client id" option for in houses?

**Comment by dex-89** [comment]
> @dex-89, excuse my ignorance, but what is this "client id" option for in houses?

Client id - House id from original (Cipsoft's) Tibia client

**Comment by luanluciano93** [comment]
Wouldn't another name be more understandable?

**Comment by dex-89** [comment]
> @dex-89 Wouldn't another name be more understandable?

Possibly yes, but this is a very old project of mine that I no longer work on. Write to @dudantas about this

**Comment by luanluciano93** [comment]
Can I create a PR with just the Max beds part? Once the canary supports "client id", we can add...

**Review by dudantas — APPROVED** [review]

**Pull request merged**